opentelemetry/src/main/scala/zio/telemetry/opentelemetry/baggage/Baggage.scala

```diff
@@ -114,39 +114,44 @@ trait Baggage { self =>
 
 object Baggage {
 
-  def live: URLayer[ContextStorage, Baggage] =
+  def live(logAnnotated: Boolean = false): URLayer[ContextStorage, Baggage] =
```

Comment:
My biggest concern is how to encode several different capabilities. For Baggage, we likely wouldn't add more, but for Tracing, we're going to have the following:

- ZIO log annotations as Span attributes
- ZIO logging as Span log events

In this case, I chose to go the most straightforward way and added one more argument to the live layer constructor to dispatch on it. It leads to one additional boolean check in the runtime. Of course, for Baggage, we can have separate `Baggage` trait implementations because we have only one capability. In the case of Tracing, we might want to give a user an option to combine these capabilities the way he/she wants, and for this, the approach of having separate implementations of the Tracing trait wouldn't work.